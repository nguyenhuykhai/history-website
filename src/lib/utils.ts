import { clsx, type ClassValue } from "clsx"
import { twMerge } from "tailwind-merge"

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs))
}

export function formatDate(date: string | null) {
  if (!date) return "";
  const [year, month, day] = date.split("-");
  return `${day}/${month}/${year}`;
}
