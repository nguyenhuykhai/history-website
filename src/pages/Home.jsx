import React from "react";
import NewsSection from "../components/NewsSection";
import LeaderSection from "../components/LeaderSection";
import SuggestionsSection from "../components/SuggestionsSection";

const Home = () => {
  return (
    <div>
      <main className="container mx-auto">
        <NewsSection />
        <SuggestionsSection />
        <LeaderSection />
      </main>
    </div>
  );
};

export default Home;