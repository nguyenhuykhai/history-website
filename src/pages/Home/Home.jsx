import React from "react";
import NewsSection from "./components/NewsSection";
import LeaderSection from "./components/LeaderSection";
import SuggestionsSection from "./components/SuggestionsSection";
import OperationSection from "./components/OperationSection";
import BannerSection from "./components/BannerSection";

const Home = () => {
  return (
    <div>
      <main className="container mx-auto">
        <BannerSection />
        <NewsSection />
        {/* Góp ý xây dựng Đảng */}
        <SuggestionsSection />
        {/* Ban lãnh đạo */}
        <LeaderSection />
        {/* Tổ chức Đảng */}
        <OperationSection />
      </main>
    </div>
  );
};

export default Home;
