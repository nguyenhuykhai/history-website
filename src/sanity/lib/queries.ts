import { defineQuery } from "groq";

export const TERMS_QUERY = defineQuery(
  `*[_type == "terms"] | order(_createdAt asc) {
  _id,
  name,
  description,
  data[] -> {
    name, _id, title, image
  },
  _createdAt,
  _updatedAt,
 }`
);

export const NEWS_QUERY = defineQuery(
  `*[_type == "news" && defined(slug.current)] {
  slug,
  title,
  image,
  category,
  author -> {
    name, _id, title
  },
  createdAt,
  description
 }`
);
