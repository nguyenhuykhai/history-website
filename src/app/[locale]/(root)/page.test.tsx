import {
  expectedHomeMetadata,
  mockNews,
  mockTerms,
  mockUnits,
} from "@/__test__/__mocks__/mockData";
import Home, { metadata } from "@/app/[locale]/(root)/page";
import { client } from "@/sanity/lib/client";
import { notFound } from "next/navigation";

jest.mock("next/navigation", () => ({
  notFound: jest.fn(),
}));

jest.mock("@/sanity/lib/client", () => ({
  client: {
    fetch: jest.fn(),
  },
}));

jest.mock("@/sanity/lib/live", () => ({
  defineLive: jest.fn(() => ({
    sanityFetch: jest.fn(),
    SanityLive: jest.fn(() => (
      <div data-testid="sanity-live">SanityLive Mock</div>
    )),
  })),
}));

describe("Home Page", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("calls notFound if terms data is missing", async () => {
    (client.fetch as jest.Mock)
      .mockResolvedValueOnce(null)
      .mockResolvedValueOnce(mockNews)
      .mockResolvedValueOnce(mockUnits);

    await Home();

    expect(notFound).toHaveBeenCalled();
  });

  it("calls notFound if news data is missing", async () => {
    (client.fetch as jest.Mock)
      .mockResolvedValueOnce(mockTerms)
      .mockResolvedValueOnce(null)
      .mockResolvedValueOnce(mockUnits);

    await Home();

    expect(notFound).toHaveBeenCalled();
  });

  it("calls notFound if units data is missing", async () => {
    (client.fetch as jest.Mock)
      .mockResolvedValueOnce(mockTerms)
      .mockResolvedValueOnce(mockNews)
      .mockResolvedValueOnce(null);

    await Home();

    expect(notFound).toHaveBeenCalled();
  });

  it("calls notFound if all data is missing", async () => {
    (client.fetch as jest.Mock)
      .mockResolvedValueOnce(null)
      .mockResolvedValueOnce(null)
      .mockResolvedValueOnce(null);

    await Home();

    expect(notFound).toHaveBeenCalled();
  });

  it("renders metadata correctly", () => {
    expect(metadata).toEqual(expectedHomeMetadata);
  });
});