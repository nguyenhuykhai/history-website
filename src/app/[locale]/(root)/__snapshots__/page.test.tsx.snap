// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders correctly with fetched data 1`] = `<Home />`;
